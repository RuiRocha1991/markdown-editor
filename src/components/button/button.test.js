'use strict'

import React from 'react'
import Button from './index'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

it('Should Button default to matcg snapshot',() => {
  const tree = renderer.create(
    <Button onClick={() => null}>
      Click me
    </Button>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('Should Button success to matcg snapshot',() => {
  const tree = renderer.create(
    <Button onClick={() => null} kind='success'>
      Click me success
    </Button>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('Should Button danger to matcg snapshot',() => {
  const tree = renderer.create(
    <Button onClick={() => null} kind='danger'>
      Click me danger
    </Button>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('Should Button with prop Kind="success" should has class "-success"',() => {
  const wrapper = shallow( <Button onClick={() => null} kind='success'>
  Click me success
</Button>)

  expect(wrapper.hasClass('-success')).toBe(true)
})

it('Should Button with prop Kind="danger" should has class "-danger"',() => {
  const wrapper = shallow( <Button onClick={() => null} kind='danger'>
  Click me success
</Button>)

  expect(wrapper.hasClass('-danger')).toBe(true)
})
