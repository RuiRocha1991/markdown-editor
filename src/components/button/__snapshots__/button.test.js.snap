// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Button danger to matcg snapshot 1`] = `
<button
  className="button -danger"
  onClick={[Function]}
>
  Click me danger
</button>
`;

exports[`Should Button default to matcg snapshot 1`] = `
<button
  className="button"
  onClick={[Function]}
>
  Click me
</button>
`;

exports[`Should Button success to matcg snapshot 1`] = `
<button
  className="button -success"
  onClick={[Function]}
>
  Click me success
</button>
`;
